### PR TITLE
refactor styler tests

### PR DESCRIPTION
**Pull request by attack68**
splits `test_style.py` to `style/test_style.py` and `style/test_builtins.py`, with the simple classification of being related to `core` functionality or added `extensions`.

### CODE REVIEW

File: pandas/tests/io/formats/style/test_builtins.py [hunk 1]
```diff
@@ -0,0 +1,656 @@
...
+    def test_tooltip_class(self):
+        # GH 21266
+        df = DataFrame(data=[[0, 1], [2, 3]])
+        s = (
+            Styler(df, uuid_len=0)
+            .set_tooltips(DataFrame([["tooltip"]]))
+            .set_tooltips_class(name="other-class", properties=[("color", "green")])
+            .render()
+        )
+        assert "#T__ .other-class {\n  color: green;\n" in s
```
Comment (jreback) [inline pandas/tests/io/formats/style/test_builtins.py:553]: split to more modules, e.g. tooltips to separate, the highlite* to a separate on, and rename this to align

File: pandas/tests/io/formats/style/test_builtins.py [hunk 2]
```diff
@@ -0,0 +1,656 @@
...
+            Styler(df, uuid_len=0)
+            .set_tooltips(DataFrame([["tooltip"]]))
+            .set_tooltips_class(name="other-class", properties="color:green;color:red;")
+            .render()
+        )
+        assert "#T__ .other-class {\n  color: green;\n  color: red;\n}" in s
+
+
+@td.skip_if_no_mpl
+class TestStylerMatplotlibDep:
```
Comment (jreback) [inline pandas/tests/io/formats/style/test_builtins.py:567]: move to this to test_matplotlib.py

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/formats/style/test_builtins.py (hunk 1 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_builtins.py]: done

File: pandas/tests/io/formats/style/test_builtins.py (hunk 2 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_builtins.py]: could not find. used existing file `tests/plotting/test_style.py` instead. `test_misc.py` also seemed a good alternative.

Comment (jreback) [reply pandas/tests/io/formats/style/test_builtins.py]: these are all *new* modules.

File: pandas/tests/plotting/test_style.py [hunk 3]
```diff
@@ -1,6 +1,11 @@
+import numpy as np
```
Comment (jreback) [inline pandas/tests/plotting/test_style.py:1]: why did you move here? these should all be with the styler tests.

**Review by jreback — CHANGES_REQUESTED** [review]

File: pandas/tests/io/formats/style/test_builtins.py (hunk 2 above)
Comment (attack68) [reply pandas/tests/io/formats/style/test_builtins.py]: duh.. my bad, assumed you were pointing to existing file

File: pandas/tests/plotting/test_style.py (hunk 3 above)
Comment (attack68) [reply pandas/tests/plotting/test_style.py:1]: very obvious in hindsight what you intended.

**Review by jreback — APPROVED** [review]

**Comment by jreback** [comment]
great thanks @attack68
